### ISSUE TITLE
[1.5.0] Error building tar and zip file

### ISSUE THREAD

**Issue by suyograo** [elasticsearch/logstash#2757]
Tried building artifacts on 1.5 -- does not work
```
git clone logstash
git checkout 1.5
rake artifact:tar
```

```
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException
Error Bundler::InstallError, retrying 1/10
Error Errno::EBADF, retrying 2/10
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]
[ERROR]   The project rubygems:logstash-input-log4j:0.1.3 (/home/ubuntu/logstash/vendor/bundle/jruby/1.9/specifications/logstash-input-log4j-0.1.3-java.gemspec) has 3 errors
[ERROR]     Unresolveable build extension: Plugin de.saumya.mojo:gem-extension: or one of its dependencies could not be resolved: Failure to find de.saumya.mojo:gem-extension:jar: in http://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced -> [Help 2]
[ERROR]     Unknown packaging: gem
[ERROR]     'build.plugins.plugin.version' for de.saumya.mojo:gem-maven-plugin must be a valid version but is ''.
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException
Error Bundler::InstallError, retrying 3/10
Error Errno::EBADF, retrying 4/10
````

This is seen in our Jenkins too and issue #2756

**Comment by ph** [comment]
:hotsprings:

**Comment by suyograo** [comment]
This however works in 2.0 branch

**Comment by suyograo** [comment]
https://github.com/logstash-plugins/logstash-input-kafka/issues/21

**Comment by ph** [comment]
After doing a successful `rake artifact:tar` on the master branch this is the difference between the master and the 1.5 branch.
```diff
diff --git a/Gemfile.jruby-1.9.lock b/Gemfile.jruby-1.9.lock
index 757f612..c20e3a3 100644
--- a/Gemfile.jruby-1.9.lock
+++ b/Gemfile.jruby-1.9.lock
@@ -1,7 +1,7 @@
 PATH
   remote: .
   specs:
-    logstash-core (2.0.0.dev-java)
+    logstash-core (1.5.0.rc1.1-java)
       cabin (>= 0.7.0)
       clamp
       file-dependencies (= 0.1.6)
@@ -62,12 +62,12 @@ GEM
     diff-lcs (1.2.5)
     docile (1.1.5)
     edn (1.0.6)
-    elasticsearch (1.0.8)
-      elasticsearch-api (= 1.0.7)
-      elasticsearch-transport (= 1.0.7)
-    elasticsearch-api (1.0.7)
+    elasticsearch (1.0.6)
+      elasticsearch-api (= 1.0.6)
+      elasticsearch-transport (= 1.0.6)
+    elasticsearch-api (1.0.6)
       multi_json
-    elasticsearch-transport (1.0.7)
+    elasticsearch-transport (1.0.6)
       faraday
       multi_json
     equalizer (0.0.9)
diff --git a/lib/logstash/version.rb b/lib/logstash/version.rb
index 36f2bad..d707741 100644
--- a/lib/logstash/version.rb
+++ b/lib/logstash/version.rb``

**Comment by suyograo** [comment]
Its very strange that I can create deb and rpm packages with valid jars on a clean git clone, but not tar or zip

```
nstalling fpm >= 0 because the build process needs it.
[bootstrap] Fetching and installing gem: fpm (>= 0)
Completed: /home/ubuntu/1.5/logstash/rakelib/../build/logstash_1.5.0.rc1.1-1_all.deb
```

**Comment by ph** [comment]
Only difference is the `logstash-core 2.0.0-dev-jav` isn't on rubygems but the other is.

**Comment by suyograo** [comment]
Same issue with elasticsearch-output

```
ERROR]   The project rubygems:logstash-output-elasticsearch:0.1.18 (/home/ubuntu/logstash/vendor/bundle/jruby/1.9/specifications/logstash-output-elasticsearch-0.1.18-java.gemspec) has 3 errors
[ERROR]     Unresolveable build extension: Plugin de.saumya.mojo:gem-extension: or one of its dependencies could not be resolved: Failure to find de.saumya.mojo:gem-extension:jar: in http://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced -> [Help 2]
[ERROR]     Unknown packaging: gem
[ERROR]     'build.plugins.plugin.version' for de.saumya.mojo:gem-maven-plugin must be a valid version but is ''.
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException
```

**Comment by suyograo** [comment]
Diff b/w master and 1.5 
https://gist.github.com/suyograo/689b847a742bf9ff6a7c

**Comment by purbon** [comment]
My best guess is that the root cause of that issue is that in the 1.5 branch we're asking to install the plugins, while in the master branch we're not (doing that using the rake task during the package build).

In the 1.5 branch, if you clean up lines https://github.com/elasticsearch/logstash/blob/1.5/Gemfile#L6-L105 between 6 and 105, then the jars are installed properly.

```
➜  logstash git:(fix/jars) ✗ find ./vendor | grep elasticsearch | grep jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/com/spatial4j/spatial4j/0.4.1/spatial4j-0.4.1jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/logstash-output-elasticsearch_jars.rb
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/antlr/antlr-runtime/3.5/antlr-runtime-3.5jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/apache/lucene/lucene-analyzers-common/4.10.2/lucene-analyzers-common-4.10.2.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/apache/lucene/lucene-core/4.10.2/lucene-core-4.10.2.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/apache/lucene/lucene-grouping/4.10.2/lucene-grouping-4.10.2.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/apache/lucene/lucene-highlighter/4.10.2/lucene-highlighter-4.10.2.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/apache/lucene/lucene-join/4.10.2/lucene-join-4.10.2.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/apache/lucene/lucene-memory/4.10.2/lucene-memory-4.10.2.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/apache/lucene/lucene-misc/4.10.2/lucene-misc-4.10.2.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/apache/lucene/lucene-queries/4.10.2/lucene-queries-4.10.2.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/apache/lucene/lucene-queryparser/4.10.2/lucene-queryparser-4.10.2.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/apache/lucene/lucene-sandbox/4.10.2/lucene-sandbox-4.10.2.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/apache/lucene/lucene-spatial/4.10.2/lucene-spatial-4.10.2.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/apache/lucene/lucene-suggest/4.10.2/lucene-suggest-4.10.2.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/elasticsearch/elasticsearch/1.4.0/elasticsearch-1.4.0.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/ow2/asm/asm/4.1/asm-4.1.jar
./vendor/bundle/jruby/1.9/gems/logstash-output-elasticsearch-0.1.18-java/lib/org/ow2/asm/asm-commons/4.1/asm-commons-4.1.jar
➜  logstash git:(fix/jars) ✗ find ./vendor | grep elasticsearch | grep jar
```
I see the need of having this lines, but could we add them in a post processing call during the artifacts build? I guess this would solve this issue.

**Comment by purbon** [comment]
Actually the task ```rake artifact:prepare``` ( https://github.com/elasticsearch/logstash/blob/1.5/rakelib/artifacts.rake#L46) that is calling bootstrap and plugins:install-default will make sure to add the requested plugins to the gemfile when releasing code. So my best take to fix this is to avoid having https://github.com/elasticsearch/logstash/blob/1.5/Gemfile#L6-L105.

**Comment by purbon** [comment]
A plausible explanation, plus probable way to fix is plotted in #2761, but my best guess is we should revert and go ahead with the release. But this should be fix by 1.5.0 GA, so we're able to fix plugin versions.
